applications/ShapeOptimizationApplication/custom_utilities/response_functions/face_angle_response_function_utility.h

```diff
@@ -0,0 +1,269 @@
+// ==============================================================================
+//  KratosShapeOptimizationApplication
+//
+//  License:         BSD License
+//                   license: ShapeOptimizationApplication/license.txt
+//
+//  Main authors:    Geiser Armin, https://github.com/armingeiser
+//
+// ==============================================================================
+
+#ifndef FACE_ANGLE_RESPONSE_FUNCTION_UTILITY_H
+#define FACE_ANGLE_RESPONSE_FUNCTION_UTILITY_H
+
+// ------------------------------------------------------------------------------
+// System includes
+// ------------------------------------------------------------------------------
+#include <iostream>
+#include <string>
+
+// ------------------------------------------------------------------------------
+// External includes
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// Project includes
+// ------------------------------------------------------------------------------
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "utilities/variable_utils.h"
+#include "custom_utilities/geometry_utilities.h"
+
+// ==============================================================================
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+
+ */
+
+class FaceAngleResponseFunctionUtility
+{
+public:
+	///@name Type Definitions
+	///@{
+
+	typedef array_1d<double, 3> array_3d;
+
+	/// Pointer definition of FaceAngleResponseFunctionUtility
+	KRATOS_CLASS_POINTER_DEFINITION(FaceAngleResponseFunctionUtility);
+
+	///@}
+	///@name Life Cycle
+	///@{
+
+	/// Default constructor.
+	FaceAngleResponseFunctionUtility(ModelPart& rModelPart, Parameters ResponseSettings)
+	: mrModelPart(rModelPart)
+	{
+		const std::size_t domain_size = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+		KRATOS_ERROR_IF(domain_size != 3) << "FaceAngleResponseFunctionUtility can only be used on 3D geometries!" << std::endl;
+
+		mMainDirection = ResponseSettings["main_direction"].GetVector();
+		const double direction_norm = norm_2(mMainDirection);
+		KRATOS_ERROR_IF(domain_size != 3) << "FaceAngleResponseFunctionUtility: 'main_direction' vector norm is 0!" << std::endl;
+		mMainDirection /= direction_norm;
+
+		double min_angle = ResponseSettings["min_angle"].GetDouble();
+		mSinMinAngle = std::sin(min_angle * M_PI / 180);
+
+		std::string gradient_mode = ResponseSettings["gradient_mode"].GetString();
+		if (gradient_mode.compare("finite_differencing") == 0)
```

Comment:
```suggestion
		if (gradient_mode == "finite_differencing")
```

easier to understand